internal/cloudapi/v2/handler.go

```diff
@@ -1125,11 +1129,25 @@ func (h *apiHandlers) getComposeMetadataImpl(ctx echo.Context, id string) error
 		})
 	}
 
-	if result.OSBuildOutput == nil || len(result.OSBuildOutput.Log) == 0 {
+	if err != nil {
+		return HTTPErrorWithInternal(ErrorComposeNotFound, err)
+	}
+
+	hasOSBuildLogs, err := h.server.workers.TestResultFieldExists(jobId, "osbuild_output.log")
+	if err != nil {
+		return HTTPErrorWithInternal(ErrorComposeNotFound, err)
+	}
+
+	if result.OSBuildOutput == nil || !hasOSBuildLogs {
 		// no osbuild output recorded for job, error
 		return HTTPError(ErrorMalformedOSBuildJobResult)
 	}
 
+	// Get the metadata field into the result as it wasn't fetched during initialization.
+	err = h.server.workers.QueryResultFields(jobId, []string{"osbuild_output.metadata"}, &result)
```

Comment:
Now, this is interesting. This method now does 3 queries instead of just 1. I wonder what will be the impact of the RTT of the 2 additional queries. Not a blocker, just food for thought.